docs/core/extensions/http-client.md

```diff
@@ -0,0 +1,248 @@
+
+When you call any of the <xref:Microsoft.Extensions.DependencyInjection.HttpClientFactoryServiceCollectionExtensions.AddHttpClient%2A> extension methods, you're adding the `IHttpClientFactory` and related services to the <xref:Microsoft.Extensions.DependencyInjection.IServiceCollection>. The `IHttpClientFactory` offers the following benefits:
+
+- Exposes the `HttpClient` class as a DI-ready type.
+- Provides a central location for naming and configuring logical `HttpClient` instances.
+- Codifies the concept of outgoing middleware via delegating handlers in `HttpClient`.
+- Provides extension methods for Polly-based middleware to take advantage of delegating handlers in `HttpClient`.
+- Manages the pooling and lifetime of underlying <xref:System.Net.Http.HttpClientHandler> instances. Automatic management avoids common Domain Name System (DNS) problems that occur when manually managing `HttpClient` lifetimes.
+- Adds a configurable logging experience (via <xref:Microsoft.Extensions.Logging.ILogger>) for all requests sent through clients created by the factory.
+
+## Consumption patterns
+
+There are several ways `IHttpClientFactory` can be used in an app:
+
+* [Basic usage](#basic-usage)
+* [Named clients](#named-clients)
+* [Typed clients](#typed-clients)
+* [Generated clients](#generated-clients)
+
+The best approach depends upon the app's requirements.
+
+### Basic usage
+
+To register the `IHttpClientFactory`, call `AddHttpClient`:
+
+:::code source="snippets/http/basic/Program.cs" range="1-12" highlight="9":::
+
+Consuming services can require the `IHttpClientFactory` as a constructor parameter with [DI][di]. The following code uses `IHttpClientFactory` to create an `HttpClient` instance:
+
+:::code source="snippets/http/basic/JokeService.cs" highlight="12,16,18,23":::
+
+Using `IHttpClientFactory` like in the preceding example is a good way to refactor an existing app. It has no impact on how `HttpClient` is used. In places where `HttpClient` instances are created in an existing app, replace those occurrences with calls to <xref:System.Net.Http.IHttpClientFactory.CreateClient%2A>.
+
+### Named clients
+
+Named clients are a good choice when:
+
+- The app requires many distinct uses of `HttpClient`.
+- Many `HttpClient` instances have different configuration.
+
+Configuration for a named `HttpClient` can be specified during registration in `ConfigureServices`:
+
+:::code source="snippets/http/named/Program.cs" range="1-23" highlight="10-20":::
+
+In the preceding code the client is configured with:
+
+- A name that's pulled from the configuration under the `"JokeHttpClientName"`.
+- The base address `https://api.icndb.com/`.
+- An `"User-Agent"` header.
+
+You can use configuration to specify HTTP client names, which is helpful to avoid misnaming clients when adding and creating. In this example, the *appsettings.json* is used to configure the HTTP client name:
```

Comment:
```suggestion
You can use configuration to specify HTTP client names, which is helpful to avoid misnaming clients when adding and creating. In this example, the *appsettings.json* file is used to configure the HTTP client name:
```